micrometer-test/src/main/java/io/micrometer/core/tck/ObservationRegistryCompatibilityKit.java

```diff
@@ -135,6 +136,50 @@ void runnableThrowingErrorShouldBeObserved() {
         verify(handler).onStop(isA(Observation.Context.class));
     }
 
+    @Test
+    void checkedRunnableShouldBeObserved() throws Exception {
+        ObservationHandler<Observation.Context> handler = mock(ObservationHandler.class);
+        when(handler.supportsContext(isA(Observation.Context.class))).thenReturn(true);
+        registry.observationConfig().observationHandler(handler);
+        Observation observation = Observation.createNotStarted("myObservation", registry);
+
+        observation.observeChecked((Observation.CheckedRunnable) () -> assertThat(registry.getCurrentObservation()).isSameAs(observation));
+        assertThat(registry.getCurrentObservation()).isNull();
+
+        verify(handler).supportsContext(isA(Observation.Context.class));
```

Comment:
thoretically we could even do a `verify in order` here to ensure that steps are done in a given order